### ISSUE TITLE
ifpack - add an interface for a hypre set function taking int ** arguments to Ifpack_Hypre

### ISSUE THREAD

**Issue by jthano** [trilinos/Trilinos#4976]
## Enhancement
@trilinos/ifpack

Ifpack_Hypre does not currently include an interface for a hypre parameter set function taking an int** argument. The only such set function in hypre is HYPRE_BoomerAMGSetGridRelaxPoints and there was a comment declaring this set function obsolete. However, this function will no longer be obsolete as it is being used for AIR AMG which can be found in the AIR branch of the development hypre repo. The AIR branch will be merged into master in the near future.

I have already added an interface for int** functions in my fork of Trilinos. I am creating this issue because the contribution guidelines state an issue should be created first and then referred to in a pull request.

**Comment by mhoemmen** [comment]
@trilinos/ifpack FYI, PR testing does not enable the Hypre TPL, so whatever testing you need will have to happen on your end.  PR tests will only ensure that the Hypre TPL is truly optional, that is, that Trilinos builds and passes tests with Hypre disabled.

**Comment by jthano** [comment]
Okay I have been using the modified ifpack_Hypre code in my fork for a while and have tested the added functionality.  What are the steps I should complete before submitting a PR. Do I need to email a signed contribution form? Or is that not required for a single contribution. Would it be acceptable to submit a PR for this change against the primary development branch?

**Comment by mhoemmen** [comment]
@jwillenbring or @trilinos/framework could comment on the contribution form issue.  Besides that, we require that PRs be submitted against the `develop` branch.  Best practice is for you to create a branch in your own fork of Trilinos, and submit the PR from that branch.  Thanks!

**Comment by jwillenbring** [comment]
> @jwillenbring or @trilinos/framework could comment on the contribution form issue. Besides that, we require that PRs be submitted against the develop branch.

@jthano How extensive is the modification? It sounds modest at a high level.

**Comment by jthano** [comment]
The modification is small, but not trivial. For every location in Ifpack_Hypre.cpp and Ifpack_Hypre.h related to the interface with hypre set functions, I added support for a new interface to a function with the signature of int (*)(HYPRE_Solver, int**). I made sure all comments for the new code match the style of the already existing code.

**Comment by jthano** [comment]
I can go ahead and submit a PR so that the code changes can be easily viewed
